packages/react-dom/src/client/ReactDOMLegacy.js

```diff
@@ -281,78 +284,92 @@ export function hydrate(
   container: Container,
   callback: ?Function,
 ): React$Component<any, any> | PublicInstance | null {
-  if (__DEV__) {
-    console.error(
-      'ReactDOM.hydrate is no longer supported in React 18. Use hydrateRoot ' +
-        'instead. Until you switch to the new API, your app will behave as ' +
-        "if it's running React 17. Learn " +
-        'more: https://reactjs.org/link/switch-to-createroot',
+  if (disableLegacyReactDOMRenderAPIs) {
+    throw new Error(
+      'ReactDOM.hydrate is no longer supported in React 19. Use hydrateRoot ' +
```

Comment:
Heads up, mentioning a specific version is inaccurate for the experimental/canary releases before React 19 is release.